chaospy/quad/collection/gauss_kronrod.py

```diff
@@ -0,0 +1,230 @@
+    [[-1.  1.]] [[-2.45 -1.    0.    1.    2.45]]
+
+Applying Gauss-Kronrod to Gauss-Hermite quadrature, for an order known to not
+exist::
+
+    >>> chaospy.generate_quadrature(5, distribution, rule="K")
+    Traceback (most recent call last):
+        ...
+    ValueError: Kronrod algorithm results in illegal coefficients;
+    Gauss-Kronrod possibly not possible for Normal(mu=0, sigma=1)
+
+Multivariate support::
+
+    >>> distribution = chaospy.J(chaospy.Uniform(0, 1), chaospy.Beta(4, 5))
+    >>> X, W = chaospy.generate_quadrature(1, distribution, rule="K")
+    >>> print(numpy.around(X, 3))
+    [[0.037 0.037 0.037 0.037 0.037 0.211 0.211 0.211 0.211 0.211 0.5   0.5
+      0.5   0.5   0.5   0.789 0.789 0.789 0.789 0.789 0.963 0.963 0.963 0.963
+      0.963]
+     [0.144 0.297 0.444 0.612 0.796 0.144 0.297 0.444 0.612 0.796 0.144 0.297
+      0.444 0.612 0.796 0.144 0.297 0.444 0.612 0.796 0.144 0.297 0.444 0.612
+      0.796]]
+    >>> print(numpy.around(W, 3))
+    [0.006 0.027 0.035 0.026 0.004 0.016 0.067 0.086 0.065 0.011 0.02  0.085
+     0.11  0.083 0.014 0.016 0.067 0.086 0.065 0.011 0.006 0.027 0.035 0.026
+     0.004]
+
+Sources
+-------
+
+Code is adapted from `quadpy`_, which adapted his code from `W. Gautschi`_.
+Algorithm for calculating Kronrod-Jacobi matrices was first published in paper
+by `D. P. Laurie`_.
+
+.. _quadpy: https://github.com/nschloe/quadpy
+.. _W. Gautschi: https://www.cs.purdue.edu/archives/2002/wxg/codes/OPQ.html
+.. _D. P. Laurie: https://doi.org/10.1090/S0025-5718-97-00861-2
+"""
+from __future__ import division
+import math
+
+import numpy
+
+from .golub_welsch import _golbub_welsch
+from ..stieltjes import generate_stieltjes
+from ..combine import combine
+
+
+def quad_gauss_kronrod(order, dist=None):
+    """
+    Generate the quadrature nodes and weights in Gauss-Kronrod quadrature.
+
+    Args:
+        order (int):
+            Quadrature order.
+        dist (chaospy.distributions.baseclass.Dist):
+            The distribution weights to be used to create higher order nodes
+            from. If omitted, use ``Uniform(-1, 1)``.
+
+    Raises:
+        ValueError:
+            Error raised if Kronrod algorithm results in negative recurrence
+            coefficients.
+
+    Example:
+        >>> abscissas, weights = quad_gauss_kronrod(6)
+        >>> print(numpy.around(abscissas, 3))
+        [[-0.991 -0.949 -0.865 -0.742 -0.586 -0.406 -0.208 -0.     0.208  0.406
+           0.586  0.742  0.865  0.949  0.991]]
+        >>> print(numpy.around(weights, 3))
+        [0.011 0.032 0.052 0.07  0.085 0.095 0.102 0.105 0.102 0.095 0.085 0.07
+         0.052 0.032 0.011]
+    """
+    if dist is None:
+        from chaospy.distributions.collection import Uniform
+        dist = Uniform(lower=-1, upper=1)
+
+    # Get the Jacobi recurrence coefficients
+    length = int(math.ceil(3 * (order+1) / 2.0))
+    _, _, coeffs_a, coeffs_b = generate_stieltjes(dist, length, retall=True)
```

Comment:
Do you perform the computation in floating point arithmetic? stieltjes is potentially unstable, resulting in a wrong scheme. Modified Chebyshev is preferred; see https://github.com/nschloe/quadpy/wiki/Creating-your-own-Gauss-quadrature-in-three-simple-steps.

Reply:
I observed that you have modified Chebyshev algorithm available. I intend to port it over as well, but it will require more effort than adding individual quadrature rules.

My implementation is calculated with floating point, yes. But it is discretized Stieltjes, by Gautschi, not the classical one by Stieltjes.

As for what is best, I usually referred to Gautschi's work. He seems to have usecases for them both the algorithms as far as I can see.

BTW in his "Selected Works: Volume 2" he makes a distinction between discretized modified Chebyshev and modified Chebyshev. Which one is the one you have implemented?